src/CaptureService/CaptureService.cpp

```diff
@@ -102,9 +103,18 @@ void CaptureService::StartEventProcessing(const CaptureOptions& capture_options)
                                               capture_start_timestamp_ns_, clock_resolution_ns_));
 }
 
-void CaptureService::FinalizeEventProcessing() {
+void CaptureService::FinalizeEventProcessing(StopCaptureReason stop_capture_reason) {
+  ProducerCaptureEvent capture_finished;
+  switch (stop_capture_reason) {
+    case StopCaptureReason::kClientStop:
+      capture_finished = CreateSuccessfulCaptureFinishedEvent();
+      break;
+    case StopCaptureReason::kMemoryWatchdog:
+      capture_finished = CreateMemoryThresholdExceededCaptureFinishedEvent();
+      break;
```

Comment:
Am I getting this right?: When we exceed the memory threshold we send a capture finished event and the client will finish the RPC - basically as if the user had stopped the capture?

Reply:
Almost. We send `CaptureFinished`, but what really finishes the capture on the client is that we return from `orbit_grpc_protos::CaptureService::Service::Capture`, causing `reader_writer_->Read(&response)` to return false (the last response was indeed `CaptureFinished`). Still, this is the same as when the client initiates the stop. The difference is that when the client initiates the stop, it does it by calling `reader_writer_->WritesDone()`, but as long as we return from `Service::Capture` this is not necessary.

Reply:
Got it. Thanks!